File: pyhon_src/msg_self/scripts/sub.py
#! /usr/bin/env python3

import rospy
from msg_self.msg import Student


def stu_cb(stu):
    rospy.loginfo("%s: i received s1, name = %s, age = %d" %(rospy.get_caller_id(), stu.name, stu.age))

def main():
    rospy.init_node("msg_sub")
    rospy.Subscriber("student", Student, stu_cb)

    rospy.spin()


if __name__ == "__main__":
    main()

File: cpp_src/msg_self/src/sub.cpp
#include <ros/ros.h>
#include <msg_self/Student.h>

void stu_cb(const msg_self::Student::ConstPtr &msg) {
    ROS_INFO("i received: %s, %u", msg->name.c_str(), msg->age);
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "msg_sub");
    ros::NodeHandle nh;

    ros::Subscriber num_sub = nh.subscribe("student", 1000, &stu_cb);

    ros::spin();
    return 0;
}